django/serializing-data.md

```diff
@@ -0,0 +1,14 @@
+# Using Django Rest Framework serializers for data serialization
+Django Rest Framework can be useful for serializing data beyond the context of an API endpoint. We find it helpful whenever we're serializing json for use in a React component, like when you need to pass GeoJSON to a map or your data objects have nested relationships.
+
+[DRF's docs](https://www.django-rest-framework.org/api-guide/serializers/) are the best place to learn about the serializer classes, but this document has examples for how we use this pattern within the DataMade stack.
+
+## Use cases
+- Serializing GeoJSON or JSON data for use in a Django template or React component that's [baked into the Django template](/django/django-react-integration.md)
+- Serializing a queryset with nested relationships
+  -  This helps with database efficiency because you can leverage `prefetch_related` and then serialize all of the related data in a cleaner way. [This blog post is a good resource to learn about that](https://hakibenita.com/django-rest-framework-slow).
+
+
+## DataMade Examples:
+1. **IL NWSS app**: We [serialized GeoJSON data and linked the data is linked to model](https://github.com/illinoisdpi/il-nwss-dashboard/pull/171).
```

Comment:
This might be a typo. Did you mean something like `We serialized GeoJSON data and linked the data to a model`?

Also I suppose it's because I'm not on this project, but this link is displaying a 404 page for me so I can't see the example. Do we have any more open repos that use this serialization? I imagine this is also the case for folks that aren't a part of CCFP.

The other option might be to put code examples into this doc, but that might be more involved than you were hoping.

Reply:
oop @xmedr good catch on the typo, my guess is that i made an edit and didn't remove all the text.

and re: 404...hmm... i'm not sure if there are other repos to share. a code example would be a nice solution to that. do you think i could get by with snippets in this doc, or would it be better to do a more in-depth tutorial with an entire example repo (sort of like [this react-leaflet tutorial](https://github.com/datamade/how-to/blob/main/mapping/react-leaflet.md) but not nearly as big or thorough)

Reply:
i think the resources you've got in there have a good bit of examples and explanations already, so you're probably okay with some code snippets here, maybe paired with a sentence of what they're trying to achieve.

it seems that second use case resource has some decent simple examples of those serializers, like the [section on regular serializers](https://hakibenita.com/django-rest-framework-slow#regular-serializer), though it frames it as being a slow process. is something like that what this doc is about? if so maybe you can borrow from those to make it a bit easier

Reply:
@xmedr 

![5-months-later](https://github.com/datamade/how-to/assets/38969506/08d7dfbd-2d0b-42c7-b35c-801cb91bf2bd)

I added the code snippets instead of linking to the private repos.

You said:

> it seems that second use case resource has some decent simple examples of those serializers, like the [section on regular serializers](https://hakibenita.com/django-rest-framework-slow#regular-serializer), though it frames it as being a slow process. is something like that what this doc is about? if so maybe you can borrow from those to make it a bit easier

No, this documentation doesn't go into this. That blog post does a good job exploring that, so I thought it was relevant to link here.